controllers/restore_controller.go

```diff
@@ -800,3 +810,124 @@ func processRestoreWait(
+		Namespace: importControllerNamespace,
+	}, configMap)
+
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			// ConfigMap doesn't exist, ImportOnly is the default
+			logger.Info("import-controller-config ConfigMap not found, ImportOnly strategy is default")
+			return true
+		}
+		// Error getting ConfigMap, assume ImportOnly for safety
+		logger.Error(err, "Error getting import-controller-config ConfigMap, assuming ImportOnly")
+		return true
+	}
+
+	// ConfigMap exists, check the autoImportStrategy value
+	strategy, exists := configMap.Data[autoImportStrategyKey]
+	if !exists || strategy == "" {
+		// autoImportStrategy not set, ImportOnly is the default
+		logger.Info("autoImportStrategy not set in ConfigMap, ImportOnly strategy is default")
+		return true
+	}
+
+	// Check if strategy is ImportOnly
+	isImportOnly := strings.EqualFold(strings.TrimSpace(strategy), importOnlyStrategy)
+	logger.Info("autoImportStrategy found", "strategy", strategy, "isImportOnly", isImportOnly)
+	return isImportOnly
+}
+
+// shouldSkipManagedClusterForAnnotation checks if a ManagedCluster should be skipped
+// when adding the immediate-import annotation
+func shouldSkipManagedClusterForAnnotation(mc *clusterv1.ManagedCluster, localClusterName string) bool {
+	// Skip local-cluster by label
+	if isLocalCluster(mc) {
+		return true
+	}
+	// Skip if name matches local cluster name
+	if localClusterName != "" && mc.Name == localClusterName {
+		return true
+	}
+	// Skip if annotation already exists with a non-empty value (e.g., "Completed")
+	if annotations := mc.GetAnnotations(); annotations != nil {
+		if value, exists := annotations[immediateImportAnnotation]; exists && value != "" {
+			return true
+		}
+	}
+	return false
+}
+
+// addImmediateImportAnnotationToManagedClusters adds the immediate-import annotation
+// to all ManagedClusters (excluding local-cluster) when ImportOnly strategy is set.
+// This ensures clusters are reimported during failover scenarios.
+func addImmediateImportAnnotationToManagedClusters(ctx context.Context, c client.Client) error {
+	logger := log.FromContext(ctx)
+
+	if !isImportOnlyStrategySet(ctx, c) {
+		logger.Info("ImportAndSync strategy is set, skipping immediate-import annotation")
+		return nil
+	}
+
+	logger.Info("ImportOnly strategy detected, adding immediate-import annotation to ManagedClusters")
+
+	localClusterName, _ := getLocalClusterName(ctx, c)
+
+	managedClusters := &clusterv1.ManagedClusterList{}
+	if err := c.List(ctx, managedClusters); err != nil {
+		logger.Error(err, "Error listing ManagedClusters")
+		return err
+	}
+
+	annotatedCount := 0
+	for i := range managedClusters.Items {
+		mc := &managedClusters.Items[i]
+
+		if shouldSkipManagedClusterForAnnotation(mc, localClusterName) {
+			continue
+		}
+
+		annotations := mc.GetAnnotations()
+		if annotations == nil {
+			annotations = make(map[string]string)
+		}
+		annotations[immediateImportAnnotation] = ""
+		mc.SetAnnotations(annotations)
```

Comment:
shouldn't mc.SetAnnotations(annotations) be called  only if annotations == nil ?